src/Data/Array.c

```diff
@@ -0,0 +1,148 @@
+#include <purescript.h>
+
+//------------------------------------------------------------------------------
+// Array creation --------------------------------------------------------------
+//------------------------------------------------------------------------------
+
+PURS_FFI_FUNC_2(Data_Array_range, _start, _end, {
+	const purs_any_int_t start = purs_any_get_int(_start);
+	const purs_any_int_t end = purs_any_get_int(_end);
+	purs_any_int_t step = start > end ? -1 : 1;
+	purs_vec_t * result = (purs_vec_t *) purs_vec_new();
+	purs_vec_reserve(result, step * (end - start) + 1);
+	purs_any_int_t i = start;
+	purs_any_int_t n = 0;
+	while (i != end) {
+		result->data[n++] = purs_any_int_new(i);
+		i += step;
+	}
+	result->data[n] = purs_any_int_new(i);
+	result->length = step * (end - start) + 1;
+	return purs_any_array_new(result);
+});
```

Comment:
NB `purs_vec_X` just calls out to vec.h (https://github.com/pure-c/pure-c/blob/master/vendor/vec.h, see also https://github.com/rxi/vec). We can add more `purs_vec_X` wrappers if u find anything useful in vec.h.